lib/src/client.dart

```diff
@@ -953,21 +953,43 @@ class Client extends MatrixApi {
     }
 
     if (getFromRooms) {
-      final room = rooms.firstWhereOrNull((Room room) =>
-          room.getParticipants().indexWhere((User user) => user.id == userId) !=
-          -1);
-      if (room != null) {
-        final user =
-            room.getParticipants().firstWhere((User user) => user.id == userId);
+      final rooms = this.rooms;
+
+      User? foundUser;
+      // check in memory first
+      for (final room in rooms) {
+        if (!room.partial) {
+          foundUser = await room.requestUser(userId,
+              ignoreErrors: true, requestProfile: false, requestState: false);
+          if (foundUser != null) break;
+        }
+      }
+
+      // If no hit, check the database
+      if (foundUser == null) {
+        for (final room in rooms) {
+          if (room.partial) {
+            foundUser = await room.requestUser(userId,
```

Comment:
This would try to request the user from server for each room which would result into 1000 failed requests if we have 1000 rooms where this user is not participating

Reply:
No, it would not, because both requestProfile and requestState are set to false. None of these requests should be doing any server requests.